src/functional_data.Rmd

```diff
@@ -0,0 +1,237 @@
+performance(p, rmse)
+```
+
+```{r}
+# Alternatively, we can simply resample (3-fold Cross-Validation)
+resample(fdalrn, tsk1, resampling = cv3, measures =  mse, show.info = FALSE)
+```
+
+Alternatively, learners that do not specifically treat functional covariates can
+be applied. In this case the temporal structure is completely disregarded, and all
+columns are treated as independent.
+
+```{r}
+# Train a normal learner on the constructed task.
+# Note that we get a message, that functionals have been converted to numerics.
+rpart.lrn = makeLearner("regr.rpart")
+m = train(learner = rpart.lrn, task = tsk1)
+m
+```
+
+## Feature Extraction
+
+In contrast to applying a learner that works on a [Task](&makeTask) containing functional features,
+the [Task](&makeTask) can be converted to a normal [&Task.md].
+This works by transforming the functional features into a 
+non-functional domain, e.g by extracting wavelets.
+
+The currently supported preprocessing functions are:
+* discrete wavelet transform
+* fast fourier transform
+* functional principal component analysis
+* multi-resolution feature extraction
+
+In order to do this, we specify methods for each functional feature in the task in a __list__.
+In this case we simply want to extract the mean from each __UVVIS__ functional and the fourier transformed features from each __NIR__ functional. Additional arguments can be passed on 
+
+```{r}
+# feat.methods specifies what to extract from which functional
+# from the first functional we extract the fourier transformation, from the second the fpca scores
+feat.methods = list("UVVIS" = extractFDAFourier(), "NIR" = extractFDAFPCA())
+
+# Either create a new task from an existing task
+extracted = extractFDAFeatures(tsk1, feat.methods = feat.methods)
+extracted
+```
+
+
+### Wavelets
+
+In this case, discrete wavelet feature transformation is applied.
+We can specify which feature extraction method is used via _method = "wavelets"_ and add additional parameters (i.e. the filter and the boundary) in the pars argument. 
+This functions returns a regression task of type regr since the raw data contained temporal structure but the transformed data does not inherit temporal structure anymore.
+For more informations on wavelets consider the documentation [wavelets](dwt).
+
+```{r, eval = FALSE}
+## Specify the feature extraction method and generate new task.
+## Here, we use the Haar filter:
+feat.methods = list("UVVIS" = extractFDAWavelets(filter = "haar"))
+task.w = extractFDAFeatures(tsk1, feat.methods = feat.methods)
+
+# Use the Daubechie wavelet with filter length 4.
+feat.methods = list("NIR" = extractFDAWavelets(filter = "d4"))
+task.wd4 = extractFDAFeatures(tsk1, feat.methods = feat.methods)
+```
+
+
+### Fourier transformation
+
+Now, we use the fourier feature transformation. Either the amplitude or the phase of the complex fourier coefficients can be used for analysis. This can be specified in the additional _fft.coeff_ argument:
```

Comment:
For newbies, it may be nice to have links to what is the amplitude and phase part of the complex fourier coefficients.